bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt

```diff
@@ -417,11 +432,32 @@ private constructor(
         private val gson = Gson()
         private var accessToken: String? = null
         private var debug = false
+        private var scheme = "https"
+        private var port = 443
+        private var trustAllCerts = false
 
         fun accessToken(accessToken: String) = apply {
             this.accessToken = accessToken
         }
 
+        fun withHttpsDisabled() = apply {
+            scheme = "http"
+        }
+
+        /**
+         * Disables certificate validation and hostname verification. Please do not use this
+         * on production environments, as it is considered bad practice. Use it solely for
+         * testing purposes.
+         */
+        fun withTrustAllCerts() = apply {
+            trustAllCerts = true
+            configureForTrustAll(okHttpClientBuilder)
+        }
+
+        fun withPort(port: Int) = apply {
```

Comment:
Can public Mastodon servers run on ports other than 443? If not, we should at least advise against using this in production, similar to the documentation of `withTrustAllCerts()`.